### PR TITLE
clients(lr): enable uses-http2, add protocol override header

### PR DESCRIPTION
**Pull request by connorjclark**
googlers: see b/157076979

### CODE REVIEW

File: lighthouse-core/config/lr-desktop-config.js [hunk 1]
```diff
@@ -15,8 +15,6 @@ const config = {
     maxWaitForLoad: 35 * 1000,
     emulatedFormFactor: 'desktop',
     throttling: desktopDense4G,
-    // Skip the h2 audit so it doesn't lie to us. See https://github.com/GoogleChrome/lighthouse/issues/6539
```
Comment (exterkamp) [inline lighthouse-core/config/lr-desktop-config.js:18]: 😍 😍 😍 😍 😍 😍 😍 😍 😍 😍 😍 😍 😍 😍 😍 😍 😍

File: lighthouse-core/test/lib/network-request-test.js [hunk 2]
```diff
@@ -71,6 +71,45 @@ describe('NetworkRequest', () => {
...
+      const record = NetworkRecorder.recordsFromLogs(devtoolsLog)[0];
+
+      expect(record.protocol).toStrictEqual('http/1.1');
+    });
+
+    it('updates protocol if Lightrider', () => {
+      const req = getRequest();
+
+      const devtoolsLog = networkRecordsToDevtoolsLog([req]);
+      global.isLightrider = true;
```
Comment (patrickhulce) [inline lighthouse-core/test/lib/network-request-test.js:95]: add an `afterEach` for `global.isLightrider = undefined`? :)

**Review by patrickhulce — COMMENTED** [review]
amazing! 🎉  😍

File: lighthouse-core/test/lib/network-request-test.js (hunk 2 above)
Comment (connorjclark) [reply lighthouse-core/test/lib/network-request-test.js:95]: already done in before each

**Review by adamraine — APPROVED** [review]
LGTM

**Pull request merged**